From 7d80079f798aec3aec530e3bef8778fb4de47953 Mon Sep 17 00:00:00 2001
From: Garrett Brown <eigendebugger@gmail.com>
Date: Wed, 17 Aug 2022 15:51:53 -0700
Subject: [PATCH] Disable -Werror
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fixes build error:

  v4l2_camera.cpp: In member function ‘bool v4l2_camera::V4L2Camera::requestImageSize(const std::vector<long long int, std::allocator<long long int> >&)’:
  v4l2_camera.cpp:351:7: error: format ‘%lu’ expects argument of type ‘long unsigned int’, but argument 5 has type ‘std::vector<long long int, std::allocator<long long int> >::size_type’ {aka ‘unsigned int’} [-Werror=format=]
         "Invalid image size; expected dimensions: 2, actual: %lu",
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         size.size());
         ~~~~~~~~~~~
  cc1plus: all warnings being treated as errors
---
 CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5ccba4f..edadfa9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -40,8 +40,6 @@ ament_target_dependencies(v4l2_camera
   "cv_bridge"
   "image_transport")
 
-target_compile_options(v4l2_camera PRIVATE -Werror)
-
 # Causes the visibility macros to use dllexport rather than dllimport,
 # which is appropriate when building the dll but not consuming it.
 target_compile_definitions(v4l2_camera PRIVATE "V4L2_CAMERA_BUILDING_LIBRARY")
-- 
2.17.1

